openapi/openapi/src/main/scala/endpoints4s/openapi/Urls.scala

```diff
@@ -48,13 +49,49 @@ trait Urls extends algebra.Urls {
       )
     )
 
-  case class DocumentedQueryStringParam(schema: Schema, isRequired: Boolean)
-  type QueryStringParam[A] = DocumentedQueryStringParam
+  def qsWithDefault[A](name: String, default: A, docs: Documentation = None)(implicit
+      value: QueryStringParam[Option[A]]
+  ): QueryString[A] =
+    DocumentedQueryString(
+      List(
+        DocumentedParameter(
+          name,
+          required = false,
+          docs,
+          schema = {
+            def withDefault(defaultValue: Option[ujson.Value]) = value.schema match {
+              case s: Schema.Object    => s.withDefault(defaultValue)
+              case s: Schema.Array     => s.withDefault(defaultValue)
+              case s: Schema.Enum      => s.withDefault(defaultValue)
+              case s: Schema.Primitive => s.withDefault(defaultValue)
+              case s: Schema.OneOf     => s.withDefault(defaultValue)
+              case s: Schema.AllOf     => s.withDefault(defaultValue)
+              case s: Schema.Reference => s
```

Comment:
Default value is not supported for `Reference` schema.
Should we better fail here, instead of doing nothing?
Anyway, `Reference` schema in the position of any kind of query string parameter (required, optional or with default value) is probably unexpected.

Reply:
Looking at #881, do you think we could follow a similar approach here and attach the default value to the `DocumentedQueryStringParam` instead of the `Schema`, to avoid this issue?

Reply:
I guess there is already agreement that #885 workaround is the right solution to this in OpenAPI 3.0.
So, I think it's best to rebase this on top of those changes, once they are stable. Do you agree @julienrf ?

Reply:
Yes, I agree

Reply:
I think using `DocumentedQueryStringParam` is still a good idea though, right? #885 just makes it possible to reliably fix a default value onto a `Schema`.

Reply:
Also, I just realized you may also want to use [`Schema#withDefinedDefault(default: Option[String]): Schema`](https://github.com/endpoints4s/endpoints4s/pull/881/files#diff-1f590f2eefffc13ac5521a40b4b8f3a3d5e539bdb5c8551508d2868c0c6c5508R975) which I just added in #881. Maybe that method should also have been added as part of #885 - sorry 😬

Reply:
Rebased to the latest master that contains #885 

>Looking at #881, do you think we could follow a similar approach here and attach the default value to the DocumentedQueryStringParam instead of the Schema, to avoid this issue?

>I think using a default value fields on DocumentedQueryStringParam is still a good idea though, right? #885 just makes it possible to reliably fix a default value onto a Schema.

As far as I understand, default value should be attached to the `Schema`, to be included in the generated OpenAPI in the end. In that sense, having it in the `DocumentedQueryStringParam` too seems redundant, since it already contains `Schema`.

> Also, I just realized you may also want to use Schema#withDefinedDefault(default: Option[String]): Schema which I just added in #881. Maybe that method should also have been added as part of #885 - sorry 😬

Yeah, I tried to add that before here, but without #885 it wasn't clean. I copied that method here, if you don't mind, as it should be merged away anyway. Now that duplicated logic is removed from `Url` trait.